### PR TITLE
Scholar into main

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Created models to save user preferences for a website PR [#87](https://github.com/Researchify/Researchify/pull/87)
Get publications using scholarly PR [#83](https://github.com/Researchify/Researchify/pull/83)
Categorizing publications PR [#81](https://github.com/Researchify/Researchify/pull/81)

# Type of change 
- [ ] Bug fix
- [X] New feature
- [X] Refactoring
- [ ] Documentation 

# Problem
#87 : No way to associate user website preferences, so added models to do that.
#83  : Before we had to scrape the results using a node package but that wasn't ideal so we decided to do it in python instead where scholarly does the scraping for us.
#81 : As per the client request, each publication should have a field indicate whether it's conference /journal and what the title of the conference/ journal.

# Solution description
See individaual PR 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes 

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
#87 Themes, templates and team model 
image
image
image

#83 See PR for more detailed information to reproduce and expected result 

#81 New publication form 
image
image

Different layout implement 
image
image

# Relevant document/ link (if any) 
#87 https://docs.google.com/document/d/1U68bJV0POrh_wDXem_hZ_PrivoVT6qZCPC8vw7Dg5uQ/edit#heading=h.p3p1n0br5dbe

# Risk
High for #81 Create and update operation for publication might not work properly if bugs exist as the publication form has been modifed 
Low for #83 and #87

### CODE REVIEW

**Review by cyr17 — APPROVED** [review]
Looks Good !

**Review by ssha0054 — APPROVED** [review]
Everything looks good from what I've seen and had no issues when running it. The only concern I have is with some of the checks that have failed, in particular the unit tests. Not really sure what the fail is for but better to have this sorted before doing the merge into main.

File: client/src/components/layout/Header.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ function Header() {
             <Navbar.Brand><Link className="header-brand" to="/dashboard">Researchify</Link></Navbar.Brand>
             <Nav className="mr-auto">
                 <Nav.Link><Link className="header-link" to="/dashboard/profile">Profile</Link></Nav.Link>
-                <Nav.Link><Link className="header-link" to="/publications/team/606bb59c22201f529db920c9">Publications </Link></Nav.Link>
+                <Nav.Link><Link className="header-link" to="/publications/team">Publications </Link></Nav.Link>
```
Comment (jfan0010) [inline client/src/components/layout/Header.js:22]: @Researchify/logiciel  has removed the links from Header.js in PR#78 and put them in Sidebar.js, but still waiting to merge though, do we need to change them in Sidebar.js as well?

**Comment by vgoh1** [comment]
> Everything looks good from what I've seen and had no issues when running it. The only concern I have is with some of the checks that have failed, in particular the unit tests. Not really sure what the fail is for but better to have this sorted before doing the merge into main.

@ssha0054 The failed static analyzer / unit tests is on the PR stage rather than the push which is weird because they would have the same content anyway. It seems to be some internal git issue, which is out of our hands. Rest assured the two commits with the failed tests is just me commenting out two lines in a python file and then putting them back in again.

File: client/src/components/layout/Header.js (hunk 1 above)
Comment (vgoh1) [reply client/src/components/layout/Header.js:22]: Can you guys merge first assuming you're already ready? We can deal with the conflict

Comment (jfan0010) [reply client/src/components/layout/Header.js:22]: Sorry that PR#78 is blocked, I guess Scholar can merge first and we can do the patch on Logiciel later.

**Pull request merged**